src/screens/Sablier/Withdraw.js

```diff
@@ -0,0 +1,222 @@
+import { ValueSelectorCard } from 'components/ValueSelectorCard';
+import { BaseText } from 'components/Typography';
+import Button from 'components/Button';
+import { Spacing } from 'components/Layout';
+import FeeLabelToggle from 'components/FeeLabelToggle';
+
+// constants
+import { SABLIER_WITHDRAW_REVIEW } from 'constants/navigationConstants';
+
+// utils
+import { getAssetDataByAddress, getAssetsAsList, isEnoughBalanceForTransactionFee } from 'utils/assets';
+import { formatUnits } from 'utils/common';
+import { buildTxFeeInfo } from 'utils/smartWallet';
+
+// selectors
+import { accountAssetsSelector } from 'selectors/assets';
+import { useGasTokenSelector } from 'selectors/smartWallet';
+import { accountBalancesSelector } from 'selectors/balances';
+
+// actions
+import { calculateSablierWithdrawTransactionEstimateAction } from 'actions/sablierActions';
+
+// services
+import { fetchStreamBalance } from 'services/sablier';
+
+import type { NavigationScreenProp } from 'react-navigation';
+import type { Dispatch, RootReducerState } from 'reducers/rootReducer';
+import type { Assets, Asset, Rates, Balances } from 'models/Asset';
+import type { Stream } from 'models/Sablier';
+
+
+type Props = {
+  calculateSablierWithdrawTransactionEstimate: (stream: Stream, amount: number, asset: Asset) => void,
+  assets: Assets,
+  supportedAssets: Asset[],
+  navigation: NavigationScreenProp<*>,
+  baseFiatCurrency: ?string,
+  rates: Rates,
+  withdrawTransactionEstimate: ?Object,
+  isCalculatingWithdrawTransactionEstimate: boolean,
+  useGasToken: boolean,
+  balances: Balances,
+};
+
+const FooterWrapper = styled.View`
+  padding: 16px 20px;
+  align-items: center;
+  width: 100%;
+`;
+
+const Withdraw = (props: Props) => {
+  const {
+    calculateSablierWithdrawTransactionEstimate,
+    navigation,
+    assets,
+    supportedAssets,
+    baseFiatCurrency,
+    rates,
+    withdrawTransactionEstimate,
+    isCalculatingWithdrawTransactionEstimate,
+    useGasToken,
+    balances,
+  } = props;
+
+  const [withdrawAmountInWei, setWithdrawAmount] = useState(0);
+  const [maxWithdraw, setMaxWithdraw] = useState(EthersBigNumber.from(0));
+  const [isFetchingMaxWithdraw, setIsFetchingMaxWithdraw] = useState(true);
+
+  const stream = navigation.getParam('stream');
+  const { id: streamId, token, recipient } = stream;
+  const assetData = getAssetDataByAddress(getAssetsAsList(assets), supportedAssets, token.id);
+
+  useEffect(() => {
+    calculateSablierWithdrawTransactionEstimate(stream, withdrawAmountInWei, assetData);
+  }, [withdrawAmountInWei]);
+
+  useEffect(() => {
+    // the exact max amount to withdraw must be fetched from the blockchain
+    // as we can only estimate due to the fact that the contract uses block.timestamp
+    fetchStreamBalance(streamId, recipient)
+      .then((streamBalance) => {
+        setMaxWithdraw(streamBalance);
+        setIsFetchingMaxWithdraw(false);
+      })
+      .catch(() => {});
```

Comment:
We catch and report the error in the `fetchStreamBalance` function. The `catch` here is just to silence eslint. And I can't make this function async or return the promise because we're in `useEffect` hook.  But I'll silence the linter in proper way by `// eslint-disable-next-line` so it's not confusing